### PR TITLE
Implement pod varlink bindings

### PR DESCRIPTION
**Pull request by jwhonce**
* Update varlink document
* Add NoContainersInPod error in go and python
* Add support for varlink pod interface
* New code passes pylint
* Fix bug in test_runner.sh

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@jwhonce could you rebase to make sure you have the SELinux fix that is breaking tests.

**Comment by baude** [comment]
please rebase

**Comment by baude** [comment]
bot, retest this please

File: API.md [hunk 1]
```diff
@@ -1332,6 +1334,10 @@ ImageNotFound means the image could not be found by the provided name or ID in l
 ### <a name="NoContainerRunning"></a>type NoContainerRunning
 
 NoContainerRunning means none of the containers requested are running in a command that requires a running container.
+### <a name="NoContainersInPod"></a>type NoContainersInPod
+
+NoContainersInPod means a pod has no containers on which to perform operation. It contains
```
Comment (TomSweeneyRedHat) [inline API.md]: nit.  suggest "perform the operation" or perhaps "perform an operation".

File: contrib/python/podman/podman/libs/pods.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+            podman.PausePod(self._ident)
+            return self._refresh(podman)
+
+    def refresh(self):
+        """Refresh status fields for this pod."""
+        with self._client() as podman:
+            return self._refresh(podman)
+
+    def remove(self, force=False):
+        """Remove all containers in the pod, the pod, return pod ident.
```
Comment (TomSweeneyRedHat) [inline contrib/python/podman/podman/libs/pods.py]: nit "the pod, the pod"?

File: contrib/python/podman/podman/libs/pods.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            return self._refresh(podman)
+
+    def stop(self):
+        """Stop all containers in the pod."""
+        with self._client() as podman:
+            podman.StopPod(self._ident)
+            return self._refresh(podman)
+
+    def top(self):
+        """Display stats for all containers exit."""
```
Comment (TomSweeneyRedHat) [inline contrib/python/podman/podman/libs/pods.py]: drop "exit" here?

File: API.md (hunk 1 above)
Comment (jwhonce) [reply API.md]: Fixed in varlink file below.

File: cmd/podman/varlink/io.podman.varlink [hunk 4]
```diff
@@ -820,6 +820,10 @@ error PodNotFound (name: string)
 # a container ID of the container that failed.
 error PodContainerError (podname: string, errors: []PodContainerErrorData)
 
+# NoContainersInPod means a pod has no containers on which to perform operation. It contains
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply cmd/podman/varlink/io.podman.varlink]: Added 'the', you can only do one operation at a time.

File: contrib/python/podman/podman/libs/errors.py [hunk 5]
```diff
@@ -25,13 +25,25 @@ def __getattr__(self, method):
...
-    pass
-
 
 class ImageNotFound(VarlinkErrorProxy):
     """Raised when Client can not find requested image."""
 
-    pass
+
+class PodNotFound(VarlinkErrorProxy):
+    """Raised when Client can not find requested image."""
```
[the remark this answers is not in the document]
Comment (jwhonce) [reply contrib/python/podman/podman/libs/errors.py]: Updated all occurrences in file.

File: contrib/python/podman/podman/libs/pods.py (hunk 2 above)
Comment (jwhonce) [reply contrib/python/podman/podman/libs/pods.py]: fixed.

File: contrib/python/podman/podman/libs/pods.py (hunk 3 above)
Comment (jwhonce) [reply contrib/python/podman/podman/libs/pods.py]: I wonder if exit was meant for another window...  :) . Gone now.

**Comment by TomSweeneyRedHat** [comment]
LGTM, assuming happy tests (which aren't looking too happy atm, looks like Travis tripped over a flake).

**Comment by baude** [comment]
bot, retest this please

**Pull request merged**